src/core/include/mp-units/ext/prime.h

```diff
@@ -472,48 +441,34 @@ constexpr auto get_first_of(const Rng& rng, UnaryFunction f)
   return get_first_of(begin(rng), end(rng), f);
 }
 
-// A configurable instantiation of the "wheel factorization" algorithm [1] for prime numbers.
-//
-// Instantiate with N to use a "basis" of the first N prime numbers.  Higher values of N use fewer trial divisions, at
-// the cost of additional space.  The amount of space consumed is roughly the total number of numbers that are a) less
-// than the _product_ of the basis elements (first N primes), and b) coprime with every element of the basis.  This
-// means it grows rapidly with N.  Consider this approximate chart:
-//
-//   N | Num coprimes | Trial divisions needed
-//   --+--------------+-----------------------
-//   1 |            1 |                 50.0 %
-//   2 |            2 |                 33.3 %
-//   3 |            8 |                 26.7 %
-//   4 |           48 |                 22.9 %
-//   5 |          480 |                 20.8 %
-//
-// Note the diminishing returns, and the rapidly escalating costs.  Consider this behaviour when choosing the value of N
-// most appropriate for your needs.
-//
-// [1] https://en.wikipedia.org/wiki/Wheel_factorization
-template<std::size_t BasisSize>
-struct wheel_factorizer {
-  static constexpr auto basis = first_n_primes<BasisSize>();
-  static constexpr auto wheel_size = product(basis);
-  static constexpr auto coprimes_in_first_wheel =
-    coprimes_up_to<num_coprimes_up_to(wheel_size, basis)>(wheel_size, basis);
-
-  [[nodiscard]] static consteval std::uintmax_t find_first_factor(std::uintmax_t n)
-  {
-    if (const auto k = detail::get_first_of(basis, [&](auto p) { return first_factor_maybe(n, p); })) return *k;
+[[nodiscard]] consteval std::uintmax_t find_first_factor(std::uintmax_t n)
+{
+  constexpr auto first_100_primes = first_n_primes<100>();
 
-    if (const auto k = detail::get_first_of(std::next(begin(coprimes_in_first_wheel)), end(coprimes_in_first_wheel),
-                                            [&](auto p) { return first_factor_maybe(n, p); }))
-      return *k;
+  for (const auto& p : first_100_primes) {
```

Comment:
Yeah, I expect so.

We "need" these because it's generally more efficient to start with trial division for a bunch of small primes before going on to methods that scale better.  Since we already had a function to compute these, I figured I'd just use it, because then it wouldn't raise questions of correctness, in the same way that a manual list of numbers might.

The compile time penalty for this many primes should be pretty small --- I'd guess 10's or maybe 100 milliseconds.  If we find that it's too much, we can:

1. Make a hardcoded array (oh, and we might as well use `std::uint16_t`, for better vectorization).
2. Write a unit test that loops over our hardcoded array, and `first_n_primes<100>()`, and makes sure they're the same.

Reply:
I don't think that there is any vectorization done while executing code in a compile-time interpretation mode.

Reply:
I don't think that there is any vectorization done while executing code in a compile-time interpretation mode.

Reply:
I did this change and also some measurements. Here are the results:

## clang

### Before PR:

real    5m8.450s
user    50m6.349s
sys     1m21.224s

### After PR:

real    6m28.540s
user    66m10.155s
sys     1m23.383s

### With static:

real    4m45.743s
user    46m18.050s
sys     1m12.484s

## gcc

### Before PR:
real    3m51.994s
user    39m37.633s
sys     1m15.279s

### After PR:
real    3m55.843s
user    40m33.021s
sys     1m19.231s

### With static:
real    3m27.275s
user    35m42.372s
sys     1m7.863s

Reply:
Maybe there are some other places where we could benefit from memoization? Or maybe we can somehow start the computation from the largest prime we've discovered so far instead of from the beginning every time?